### ISSUE TITLE
Invoice Lineitem sequence reordering not being obeyed

### ISSUE THREAD

**Issue by LarkSpurMarauder** [AGenius/Xero.Net.OAuth2#23]
I have a winforms app that reads and writes invoices using your 'sync' wrapper.
If I use this app to append a new lineitem (A) to an existing draft invoice, then after it has saved both the returned invoice and the webpage invoice shows the new lineitem as the first item, not the last item as was expected.

If I continue with this same invoice and in another single edit operation I reorder some line items and also append a new lineitem (B) then save, then I get the new appended lineitem appearing *after* the first appended lineitem (A), and the reordered items appear back where they started.

if I reorder via the web interface, then the lineitem reordering is obeyed, and changes appear within my app.

No errors are thrown during either save.
Compiled to framework 4.8.
Using Xero.NetStandard.OAuth2Client version 1.6.0
I have tried following the code to find where lineitems are packed into the http strings, but have had no luck.

Here is simple code to demonstrate the issue. Requires a 2 line draft invoice named "PC00000-TEST1".
```
            String szWhereFilter = "InvoiceNumber.StartsWith(\"PC00000-TEST1\")";
            List<Invoice> linvoice = xeroAPI.AccountingApi.Invoices(filter: szWhereFilter);
            Invoice invoice = linvoice[0];
            LineItem li0 = invoice.LineItems[0];
            LineItem li1 = invoice.LineItems[1];
            invoice.LineItems.Clear();
            invoice.LineItems.Add(li1);
            invoice.LineItems.Add(li0);
            Invoice iSaved = xeroAPI.AccountingApi.UpdateInvoice(invoice);
            bool bSuccessfulReorder = iSaved.LineItems[0].LineItemID.Equals(li1.LineItemID);
```

(Hi, and many thanks for your work!!:))

**Comment by LarkSpurMarauder** [comment]
Found the answer at [https://github.com/XeroAPI/Xero-NetStandard/issues/374](url)
tl;dr Design decision by Xero. If I want items reordered, I must pass *all* the lineitems as new lineitems.
Xero then deletes the old items, and re-adds the lineitems as new lineitems.
Fingers crossed that you were not storing lineitem GUIDs elsewhere..  :(
